src/lib/update-cors-cloud.ts

```diff
@@ -0,0 +1,112 @@
+import { ADD_CONFIG, GET_CONFIG } from './constants';
+
+function handleOrigins(originsToAdd: string[], origins: string[], remove: boolean) {
+  /* If we set remove to true, we want to filter out origins
+   * so that we don't remove any origins not intended to be removed
+   * but in fact, do remove the ones passed in
+   */
+  if (remove) {
+    return origins.filter((origin: string) => !originsToAdd.includes(origin));
+  } else {
+    /*
+     * We do not want to remove origins here and need to maintain the original ones
+     */
+    return origins.concat(originsToAdd);
+  }
+}
+interface CorsStructure {
+  maxAge: number;
+  exposedHeaders: string[];
+  acceptedHeaders: string[];
+  allowedCredentials: boolean;
+  acceptedMethods: string[];
+  acceptedOrigins: string[];
+  enabled: boolean;
+  _id: string;
+  _type: {
+    _id: string;
+    name: string;
+    collection: boolean;
+  };
+}
+
+interface CorsResponse {
+  data: {
+    results: CorsStructure[];
+  };
+}
+
+interface CorsConfigValues {
+  AM_URL: string;
+  originsToAdd: string[];
+  ssoToken: string;
+  cookieName: string;
+  remove?: boolean;
+}
+interface AMIdAndConfig {
+  id: string;
+  acceptedOrigins: string[];
+}
+export async function updateCorsConfig({
+  AM_URL,
+  originsToAdd,
+  ssoToken,
+  cookieName,
+  remove = false, // default to not remove origins
+}: CorsConfigValues): Promise<AMIdAndConfig> {
+  if (!AM_URL) {
+    return Promise.reject('You must provide an AM_URL');
+  }
+  if (!originsToAdd) {
+    return Promise.reject('You must provide a list of origins to update the cors config with');
+  }
+  if (!ssoToken) {
+    return Promise.reject(
+      'No SSO Token provided to update the cors config, exiting without a network call',
+    );
+  }
+  try {
+    /*
+     * On success, AM returns an HTTP 201 response code,
+     * and a representation of the CORS settings, in JSON format.
+     * AM generates a UUID for the configuration, returned as the value of the
+     * _id property. You can use this ID value to update or delete
+     * the configuration with additional REST calls.
+     * The new settings take effect immediately.
+     */
+    const { data: { data } }: AxiosResponse<CorsResponse> = await axios({
+      baseURL: AM_URL,
+      url: GET_CONFIG.url,
+      method: GET_CONFIG.type,
+      headers: { [cookieName]: ssoToken },
+    });
+
+    const [{ acceptedOrigins: existingOrigins }] = data.results;
+
+    const origins = handleOrigins(originsToAdd, existingOrigins, remove);
+
+    const response = await axios({
+      baseURL: AM_URL,
+      url: ADD_CONFIG.url,
+      method: ADD_CONFIG.type,
+      headers: { [cookieName]: ssoToken },
+      data: {
+        //  https://backstage.forgerock.com/docs/am/7.1/security-guide/enable-cors-support.html#add-cors-config
+        enabled: true,
+        acceptedOrigins: origins, // probably need to concat the existing origins here also
+      },
+    });
+
+    if (response.status === 201) {
```

Comment:
I understand that this is a test, but AM in general may return `200 "OK"` status, if the configuration already exists, and is updated successfully. 
`201 "CREATED"` status is returned, when the config didn't exist before and was successfully created.

Reply:
To your point, the status code was wrong anyways then, so I fixed this.